tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/effects/debugger_effects.ts

```diff
@@ -653,6 +657,88 @@ export class DebuggerEffects {
     );
   }
 
+  /**
+   * Listens to graph-op focus events.
+   *
+   * Load graph op info from the /graphs/op_info route when necessary.
+   *
+   * @returns the IDs of the stack frames of the op's creation for downstream
+   * consumption.
+   */
+  private loadGraphOpInfo(): Observable<string[]> {
+    return this.actions$.pipe(
+      ofType(graphOpFocused),
+      withLatestFrom(
+        this.store.select(getActiveRunId),
+        this.store.select(getLoadingGraphOps)
+      ),
+      filter(([actionData, runId, loadingOps]) => {
+        const {graph_id, op_name} = actionData;
+        return (
+          runId !== null &&
+          (loadingOps[graph_id] === undefined ||
+            loadingOps[graph_id][op_name] === undefined ||
+            loadingOps[graph_id][op_name] === DataLoadState.FAILED)
+        );
+      }),
+      tap(([actionData]) =>
+        this.store.dispatch(graphOpInfoRequested(actionData))
+      ),
+      mergeMap(([actionData, runId]) => {
+        const {graph_id, op_name} = actionData;
+        return this.dataSource.fetchGraphOpInfo(runId!, graph_id, op_name).pipe(
+          map((graphOpInfoResponse) => {
+            this.store.dispatch(graphOpInfoLoaded({graphOpInfoResponse}));
+            return graphOpInfoResponse.stack_frame_ids;
+          })
+        );
+        // TODO(cais): Add catchError() to pipe.
+      })
+    );
+  }
+
+  private loadGraphOpStackFrames(
+    prevStream$: Observable<string[]>
+  ): Observable<void> {
+    return prevStream$.pipe(
+      withLatestFrom(
+        this.store.select(getActiveRunId),
```

Comment:
In case the `activeRunId` might've changed since the `loadGraphOpInfo` stream started, should we make `loadGraphOpInfo` return an object that includes the non-null runId?

Reply:
This is unlikely, but for safety, I did what you suggested.